### ISSUE TITLE
Multiple copies of Nadu's ability don't have independent trigger limits

### ISSUE THREAD

**Issue by Grath** [magefree/mage#12484]
Per [MH3 Release Notes](https://magic.wizards.com/en/news/feature/modern-horizons-3-release-notes), each copy of Nadu's ability needs to count the trigger limit independently (also applicable if you flicker Nadu to refresh the ability):

> If you somehow control more than one Nadu, each permanent you control will have that many instances of the granted ability. These abilities are not redundant. For example, if you control two Nadus and one becomes the target of a spell or ability, each of its granted abilities will trigger. If that same Nadu becomes the target of a spell or ability again later in the turn, each of its granted abilities will trigger again, as each of them have triggered only once so far this turn.
> If Nadu leaves the battlefield and returns to the battlefield in the same turn, or if Nadu leaves and another Nadu appears, the triggered ability granted by the new one is different than the triggered ability granted by the old one. The one granted by the new Nadu may trigger twice for each creature you control, even if the ability granted by the old Nadu already triggered twice for those creatures that turn. (Note that creatures you control won't have the ability granted by the old one anymore, and if you control multiple Nadus, you won't have the opportunity to take actions before the "legend rule" gets you down to one Nadu.)

**Comment by Grath** [comment]
[[Nadu, Winged Wisdom]]

**Comment by Susucre** [comment]
Uhm so the setup of keys for setting the trigger count is not quite right:

![image](https://github.com/magefree/mage/assets/34709007/385a9b89-850e-45c3-a9e6-403e3f90f647)

This is strange, because the comment on the CardUtil's method seems to indicate it should work properly on zcc change (i.e. different key):

![image](https://github.com/magefree/mage/assets/34709007/8eb07ee0-578f-4453-8baf-d7533f08e48d)

**Comment by Susucre** [comment]
Maybe the issue is with `getOriginalId()` instead of `getId()`?
Not actually sure what is the purpose of `originalId` (that was already the setup before adding support for Nadu). And seems i'm not alone on the sentiment it feels weird:

![image](https://github.com/magefree/mage/assets/34709007/2eedc0dc-066d-4cfd-badb-100d795a230f)

**Comment by JayDi85** [comment]
OriginalId used for linked abilities (it allows to check is it same ability or from another card/object). OriginalId is same for whole ability lifecycle, even after activate (example: multiple shield instances will have same originalId, but unique id — same for multiple activates of the same ability). Xmage creates full copy of the ability with new id and old original id on each activate. 

Maybe some copy code miss of the newOriginalId or assignNewId calls. Or maybe it’s a breaking fix:
* if you assign new originalId on new card then will lost linked abilities/effects (effects with getOriginalId will be broken);
* if you do NOT assign new originalId on new card then will lost unique instances and get bugs like current issue;

![IMG_0549](https://github.com/magefree/mage/assets/8344157/77da029e-36e1-4135-86bc-9c06d2f91c0f)
